native-image-support/src/main/resources/META-INF/native-image/com.google.cloud/google-cloud-graalvm-support/proxy-config.json

```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "interfaces": [
+      "jnr.unixsocket.Native$LibC",
+      "jnr.ffi.provider.LoadedLibrary"
+    ]
+  },
+  {
+    "interfaces": [
+      "jnr.enxio.channels.Native$LibC",
+      "jnr.ffi.provider.LoadedLibrary"
+    ]
+  }
```

Comment:
What is this config needed for? Or in other terms, what issues do we run into when proxy-config.json is absent?

Reply:
cloud-sql-jdbc-socket-factory use jnr-unixsocket to support unix domain socket, and jnr will use a series of dynamic proxies to automatically generate a set of jni implementations to native code.

On the other hand, like I said before, I don't know how to register dynamic proxy in graalvm Feature, if possible it should not be written directly in file.

Reply:
I see, thanks for the explanation! Yeah it would be ideal to use `Feature` but it may be difficult at the moment. It also looks like the [documentation for dynamic proxy](https://www.graalvm.org/reference-manual/native-image/Limitations/#dynamic-proxy) mentions only a configuration file. Since it is a small config, I'm okay with having the json.